frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2886 @@
+                            left: o.left
+                        });
+                    });
+
+                    $(self.$refs.container).find(".graph-note-label").each(function() {
+                        var o = plot.pointOffset({x: $(this).data("points")[0], y: $(this).data("points")[1]});
+
+                        $(this).css({
+                            left: o.left
+                        });
+                    });
+                }
+            },
+            template: '<div class="cly-vue-time-graph-w">\
+                            <div ref="container" class="graph-container"></div>\
+                            <div class="cly-vue-graph-no-data" v-if="!hasData">\
+                                <div class="inner">\
+                                    <div class="icon"></div>\
+                                    <div class="text">{{i18n("common.graph.no-data")}}</div>\
+                                </div>\
+                            </div>\
+                        </div>'
+        }
+    ));
+
+    Vue.component("cly-graph-w", countlyBaseComponent.extend(
+        // @vue/component
+        {
+            mixins: [
+                _mixins.i18n
+            ],
+            props: {
+                dataPoints: {
+                    required: true,
+                    type: Object,
+                    default: function() {
+                        return {};
+                    }
+                },
+                graphType: { required: false, type: String, default: "bar" },
+                frozen: {default: false, type: Boolean},
+                configOptions: { required: false, default: null, type: Object }
+            },
+            data: function() {
+                return {
+                    options: JSON.parse(JSON.stringify(this.configOptions))
+                };
+            },
+            computed: {
+                hasData: function() {
+                    return !!this.dataPoints;
+                }
+            },
+            watch: {
+                dataPoints: function() {
+                    this.refresh();
+                },
+                graphType: function() {
+                    this.refresh();
+                },
+                frozen: function(newValue) {
+                    if (!newValue) {
+                        this.refresh();
+                    }
+                }
+            },
+            mounted: function() {
+                this.refresh();
+            },
+            methods: {
+                refresh: function() {
+
+                    if (this.frozen || $(this.$refs.container).is(":hidden") || !this.hasData) {
+                        // no need to refresh if hidden
+                        return;
+                    }
+
+                    countlyCommon.drawGraph(this.dataPoints,
+                        $(this.$refs.container),
+                        this.graphType,
+                        this.options);
+                }
+            },
```

Comment:
Codacy found an issue: [Multiline support is limited to browsers supporting ES5 only.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)